From 8083e2709308ef1548d21e3c3a46699d4794b876 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Tue, 8 Oct 2024 17:07:54 +0200
Subject: [PATCH 725/954] feat(st-clock): do not disable clock when M33 is
 TDCID
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

NOT UPSTREAMABLE

This path can be seen as a workaround. On the STM32MP2, when the M33 is
TDCID, TF-A must not disable any clocks, as some may be owned to CPU2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 74c7d5d3734c794e9ce1272c8cf865d6327186e4 feat(st-clock): do not disable clock when M33 is TDCID]
Change-Id: I199ed46df85b6d5bb7f2bf933185e5fc2ce0c925
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/409878
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 drivers/st/clk/clk-stm32-core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/st/clk/clk-stm32-core.c b/drivers/st/clk/clk-stm32-core.c
index f5882da8f2..bc1e0aa7cc 100644
--- a/drivers/st/clk/clk-stm32-core.c
+++ b/drivers/st/clk/clk-stm32-core.c
@@ -502,11 +502,14 @@ int _clk_stm32_enable(struct stm32_clk_priv *priv, int id)
 
 void clk_stm32_disable_call_ops(struct stm32_clk_priv *priv, uint16_t id)
 {
+/* Do not disable clock when not TDCID */
+#if !STM32MP_M33_TDCID
 	const struct stm32_clk_ops *ops = _clk_get_ops(priv, id);
 
 	if (ops->disable != NULL) {
 		ops->disable(priv, id);
 	}
+#endif
 }
 
 static void _clk_stm32_disable_core(struct stm32_clk_priv *priv, int id)
-- 
2.53.0

